test/performance/io/lowlevel_stream_benchmark.cpp

```diff
@@ -98,11 +99,97 @@ void read_all(benchmark::State & state)
 #endif
 }
 
+template <tag id>
+void write_all(benchmark::State & state)
+{
+    /* prepare file for writing */
+    seqan3::test::tmp_filename filename{"foo"};
+    std::ofstream os{filename.get_path(), std::ios::binary};
+
+    // sequence to write:
+    std::vector<char> cont_rando = seqan3::test::generate_sequence<char>(10'000, 0, 0);
+
+    /* start benchmark */
+    if constexpr (id == tag::std_stream_it)
+    {
+        for (auto _ : state)
+        {
+            std::ofstream os{filename.get_path(), std::ios::binary};
+            std::ostream_iterator<char> it{os};
+
+            for (auto chr : cont_rando)
+                *it = chr;
+        }
+    }
+    else if constexpr (id == tag::std_streambuf_it)
+    {
+        for (auto _ : state)
+        {
+            std::ofstream os{filename.get_path(), std::ios::binary};
+            std::ostreambuf_iterator<char> it{os};
+
+            for (auto chr : cont_rando)
+                *it = chr;
+        }
+    }
+    else if constexpr (id == tag::seqan3_streambuf_it)
+    {
+        for (auto _ : state)
+        {
+            std::ofstream os{filename.get_path(), std::ios::binary};
+            seqan3::detail::fast_ostreambuf_iterator<char> it{*os.rdbuf()};
+
+            for (auto chr : cont_rando)
+                *it = chr;
+        }
+    }
+#ifdef SEQAN3_HAS_SEQAN2
+    else if constexpr (id == tag::seqan2_stream_it)
+    {
+        for (auto _ : state)
+        {
+            std::ofstream os{filename.get_path(), std::ios::binary};
+            auto it = seqan::Iter<std::ofstream, seqan::StreamIterator<seqan::Output>>{os};
+
+            for (auto chr : cont_rando)
+                *it = chr;
+        }
+    }
+#endif
+}
+
+void seqan3_write_range(benchmark::State & state)
+{
+    /* prepare file for writing */
+    seqan3::test::tmp_filename filename{"foo"};
+    std::ofstream os{filename.get_path(), std::ios::binary};
+
+    // sequence to write:
+    std::vector<char> cont_rando = seqan3::test::generate_sequence<char>(10'000, 0, 0);
+
+    for (auto _ : state)
+    {
+        std::ofstream os{filename.get_path(), std::ios::binary};
+        seqan3::detail::fast_ostreambuf_iterator<char> it{*os.rdbuf()};
+
+        it.write_range(cont_rando);
+    }
+}
```

Comment:
why not integrate this with the rest? Just add an extra tag `tag::seqan3_streambuf_range`!

Reply:
Because it is not a direct comparison to std and seqan2 and also not available for the input streambuf iterator?